components/butterfly/src/server/mod.rs

```diff
@@ -703,18 +711,23 @@ impl Server {
         // NOTE: This sucks so much right here. Check out how we allocate no matter what, because
         // of just how the logic goes. The value of the trace is really high, though, so we suck it
         // for now.
-        let trace_member_id = member.id.clone();
+        let member_id = member.id.clone();
         let trace_incarnation = member.incarnation;
         let trace_health = health.clone();
 
         if self.member_list.insert(member, health) || incremented_incarnation {
             trace_it!(
                 MEMBERSHIP: self,
                 TraceKind::MemberUpdate,
-                trace_member_id,
+                member_id,
                 trace_incarnation,
                 trace_health
             );
+
+            // Purge "heat" information for a member that's gone
+            if member_id != self.member_id() && health == Health::Departed {
```

Comment:
This is _almost_ identical to the code in `insert_member`. I wonder if it's possible to factor out the commonality as this sort of duplication seems prone to bugs. I don't just mean for the added code, but for the two functions more generally.

Reply:
Good point... I'm not clear on if there's actually a meaningful distinction between these functions; the names don't give it away, at least. This would probably be good to tackle in a follow-on PR.